### PR TITLE
Introduce a new dictionary that will be used to replace image in kustomize without having to add image_push targets

### PR DESCRIPTION
**Pull request by arturo-skydio**
- Tested this in `aircam` and `skyops`
- With this change, all bazel gitops targets in `aircam` build without having to pull in python deps because the image target is no longer there

Topic: avoid-image-targets
Reviewers: daniel-benton

### CODE REVIEW

**Comment by arturo-skydio** [comment]
Reviews in this chain:
└**https://github.com/Skydio/rules_gitops/pull/9** **Introduce a new dictionary that will be used to replace image in kustomize without having to add image_push targets**

**Comment by arturo-skydio** [comment]
| # | head | base | diff | date | summary |
| - | - | - | - | - | - |
| 0 | [fe1b11e7](/Skydio/rules_gitops/commit/fe1b11e7ef1ce6cf15c4e5df82a9ecbfb75cee1e) | [c5a77cc0](/Skydio/rules_gitops/commit/c5a77cc0a0191702c1573f07b805ecc772baa312) | [diff](/Skydio/rules_gitops/compare/c5a77cc0a0191702c1573f07b805ecc772baa312..fe1b11e7ef1ce6cf15c4e5df82a9ecbfb75cee1e) | Dec 14 12:22 PM |  2 files changed, 22 insertions(+), 1 deletion(-) |
| 1 | [fb0c249f](/Skydio/rules_gitops/commit/fb0c249f6eaefe9c3bddf7601d64d71304b1444a) | [c5a77cc0](/Skydio/rules_gitops/commit/c5a77cc0a0191702c1573f07b805ecc772baa312) | [diff](/Skydio/rules_gitops/compare/fe1b11e7ef1ce6cf15c4e5df82a9ecbfb75cee1e..fb0c249f6eaefe9c3bddf7601d64d71304b1444a) | Dec 20 18:24 PM |  2 files changed, 6 insertions(+), 13 deletions(-) |
| 2 | [e8af582d](/Skydio/rules_gitops/commit/e8af582d5cdac8ca8dc3ac20cd26c806197965f0) | [c5a77cc0](/Skydio/rules_gitops/commit/c5a77cc0a0191702c1573f07b805ecc772baa312) | [diff](/Skydio/rules_gitops/compare/fb0c249f6eaefe9c3bddf7601d64d71304b1444a..e8af582d5cdac8ca8dc3ac20cd26c806197965f0) | Dec 20 18:43 PM |  1 file changed, 6 insertions(+) |

File: skylib/k8s.bzl [hunk 1]
```diff
@@ -241,6 +241,17 @@ def k8s_deploy(
             image_digest_tag = image_digest_tag,
             tags = tags,
         )
+        image_names = {}
+        if images:
+            for img_name, target_name in images.items():
+                trg = target_name.replace("//", "").split(":")
+                repo = trg[0] + "/" + trg[1]
+                image_names = {
```
Comment (daniel-benton-skydio) [inline skylib/k8s.bzl]: Doesn't this overwrite `image_names` for each image in `images.items()`?

**Review by daniel-benton-skydio — COMMENTED** [review]
I have one question regarding the `image_names` input... it seems like it'll only end up with one resolved image, even if you pass it multiple?

File: skylib/k8s.bzl (hunk 1 above)
Comment (arturo-skydio) [reply skylib/k8s.bzl]: yup, you're right, when I was doing this I forgot to go back to this and figure out how to set a list of dictionaries as an argument, bazel doesn't seem to have that type available https://bazel.build/rules/lib/attr, I think I can maneuver around it though

**Review by daniel-benton-skydio — APPROVED** [review]
LGTM.  Would be nice if there were slightly more complex types here, but I think this works

**Comment by arturo-skydio** [comment]
> LGTM. Would be nice if there were slightly more complex types here, but I think this works

yea I agree, not the best but this works for now

**Pull request merged**